src/include/duckdb/main/config.hpp

```diff
@@ -118,6 +149,8 @@ struct DBConfigOptions {
 	string autoinstall_extension_repo = "";
 	//! The maximum memory used by the database system (in bytes). Default: 80% of System available memory
 	idx_t maximum_memory = (idx_t)-1;
+	//! The maximum size of the 'temp_directory' folder when set (in bytes)
+	NumericSetting maximum_swap_space = NumericSetting();
```

Comment:
I think we can turn this back into an `optional_idx` similar to maximum memory, and handle the reading/setting of it outside of the configuration.